Framework/include/QualityControl/QualityControlFactory.h

```diff
@@ -0,0 +1,58 @@
+///
+/// \file   QualityControlFactory.h
+/// \author Piotr Konopka
+///
+
+#ifndef QC_CORE_QUALITYCONTROLFACTORY_H
+#define QC_CORE_QUALITYCONTROLFACTORY_H
+
+#include <string>
+#include <Framework/WorkflowSpec.h>
+
+namespace o2
+{
+namespace quality_control
+{
+namespace core
+{
+
+/// \brief A factory class which can generate QC topologies given a configuration file.
+///
+/// A factory class which can generate QC topologies given a configuration file (example in Framework/basic.json and
+/// Framework/example-default.json). As QC topologies will be spread on both processing chain machines and dedicated
+/// QC servers, a _local_ vs. _remote_ distinction was introduced. Tasks which are _local_ should have taskRunners
+/// placed on FLP or EPN machines and their results should be merged and checked on QC servers. The 'remote' option
+/// means, that full QC chain should be located on remote (QC) machines. For the laptop development, use 'remote' tasks
+/// and generateRemoteInfrastructure() to obtain the full topology in one go.
+///
+/// \author Piotr Konopka
+class QualityControlFactory
```

Comment:
I am not (yet) convinced by the name of the class because looking at it I would expect to be able to do getQualityControl(...) , whatever that means. Think of a BankAccountFactory, or a TaskFactory or a CheckerFactory. 

What about TopologyGenerator ?

Reply:
`TopologyGenerator` works for me, I wasn't attached to the original name neither

Reply:
Though we would use then two names for the same thing. Look at: `TopologyGenerator::generateLocalInfrastructure()`. I am not sure which one, infrastructure or topology, is better.

Reply:
`InfrastructureGenerator` sounds good as well !